app/routes/index.js

```diff
@@ -377,4 +379,39 @@ const storage = multer.diskStorage({
 			
 		}
 	})
+	app.route("/addteam").get(adminAuth, async function(req,res){
+		const teamobj=await Team.find()
+		const sz=teamobj.length
+		const cleanedobj = []
+		console.log(teamobj)
+		for(let i=0;i<sz;i++)
+		{
+			const newobj={}
+			newobj.year=teamobj[i].year
+			newobj.positions=[]
+			let j= teamobj[i].positions.length
+			for(let k =0;k<j;k++)
+			{
+				const secnewobj={}
+				secnewobj.position_name=teamobj[i].positions[k].position_name;
+				secnewobj.people=[]
+				let l = teamobj[i].positions[k].people.length;
+				console.log("iterating over ", teamobj[i].year, " and " , teamobj[i].positions[k].position_name, " with strength ", l)
+				for(let p=0;p<l;p++)
+				{
+					console.log("searching for ", teamobj[i].positions[k].people[p])
+					const user = await User.findOne({_id: teamobj[i].positions[k].people[p]})
+					console.log(user)
+					if(user!=null)
+					{
+						secnewobj.people.push(user.name)
+					}
+					
+				}
+				newobj.positions.push(secnewobj)
+			}
+			cleanedobj.push(newobj)
+		}
+		res.json(cleanedobj)
+	})
```

Comment:
I think I am having trouble understanding the code.
Why do we need so many loops? This endpoint is supposed to add a member to the team right? So you just need to find a team object in the given `year`, with the given `position_name`. I believe there are inbuilt methods in mongoose to execute such complicated queries. Do check them out, If not then you will have to change the schema so that you are able to directly fetch the required object with mongoose API. We don't want to write too many `for` loops as it makes the code incomprehensible.